pkg/providers/instancetype/types.go

```diff
@@ -309,6 +313,12 @@ func pods(_ context.Context,
 	case clusterCNI == ClusterCNITypeTerway:
 		maxENIPods := (tea.Int32Value(info.EniQuantity) - 1) * tea.Int32Value(info.EniPrivateIpAddressQuantity)
 		count = int64(maxENIPods + BaseHostNetworkPods)
+	case clusterCNI == ClusterCNITypeFlannel:
+		if maxAvailableIPCount > 0 {
```

Comment:
Based on https://help.aliyun.com/zh/ack/ack-managed-and-ack-dedicated/user-guide/work-with-flannel?spm=a2c4g.11186623.0.i2

<img width="1068" alt="image" src="https://github.com/user-attachments/assets/a65034c7-ee89-403d-9283-84f0d1365e01">

It looks like the pod CIDR is different from vswitch CIDR, which means the max pods is based on the pod CIRD.
Correct me if I understand wrong.

Reply:
When users create a ACK cluster using flannel ， the pods number on one node  are configured.
![flannelcluster](https://github.com/user-attachments/assets/2404a71d-e0b3-41ea-a621-c08d8269a3db)

Reply:
Let me check wether we can resolve this from the ack api

Reply:
I think it's supposed to be here.
https://api.aliyun.com/api/CS/2015-12-15/DescribeClusterDetail
![image](https://github.com/user-attachments/assets/cbe6ba3c-2491-4e14-a30e-d7813feb3632)

Reply:
Let me create a cluster with flannel cni, and paste the corresponding output here

Reply:
![image](https://github.com/user-attachments/assets/42e14f67-e3a2-4f2c-ad0d-391880ed062b)

Reply:
`"ContainerCIDR": "10.169.0.0/16",` does not make sense for a node's max pod count, I think we can defaulting it as 256

Reply:
> "ContainerCIDR": "10.169.0.0/16", does not make sense for a node's max pod count, I think we can defaulting it as 256

+1

Reply:
Done